pkg/server/api/svid/v1/service.go

```diff
@@ -283,8 +249,84 @@ func (s *Service) newX509SVID(ctx context.Context, param *svid.NewX509SVIDParams
 	}, nil
 }
 
-func (s *Service) NewJWTSVID(context.Context, *svid.NewJWTSVIDRequest) (*svid.NewJWTSVIDResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+func (s *Service) mintJWTSVID(ctx context.Context, protoID *types.SPIFFEID, audience []string, ttl int32) (*types.JWTSVID, error) {
+	log := rpccontext.Logger(ctx)
+
+	id, err := api.IDFromProto(protoID)
+	if err != nil {
+		log.WithError(err).Error("Failed to parse SPIFFE ID")
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	if err := idutil.ValidateTrustDomainWorkload(id, s.td); err != nil {
+		log.Errorf("Invalid SPIFFE ID: %v", err)
+		return nil, status.Errorf(codes.InvalidArgument, "invalid SPIFFE ID: %v", err)
+	}
+
+	log = log.WithField(telemetry.SPIFFEID, id.String())
+
+	if len(audience) == 0 {
+		log.Error("At least one audience is required")
+		return nil, status.Error(codes.InvalidArgument, "at least one audience is required")
+	}
+
+	token, err := s.ca.SignJWTSVID(ctx, ca.JWTSVIDParams{
+		SpiffeID: id.String(),
+		TTL:      time.Duration(ttl) * time.Second,
+		Audience: audience,
+	})
+	if err != nil {
+		log.WithError(err).Error("Failed to sign JWT-SVID")
+		return nil, status.Errorf(codes.Internal, "failed to sign JWT-SVID: %v", err)
+	}
+
+	issuedAt, expiresAt, err := jwtsvid.GetTokenExpiry(token)
+	if err != nil {
+		log.WithError(err).Error("Failed to get JWT-SVID expiry")
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &types.JWTSVID{
+		Token:     token,
+		Id:        api.ProtoFromID(id),
+		ExpiresAt: expiresAt.Unix(),
+		IssuedAt:  issuedAt.Unix(),
+	}, nil
+}
+
+func (s *Service) NewJWTSVID(ctx context.Context, req *svid.NewJWTSVIDRequest) (resp *svid.NewJWTSVIDResponse, err error) {
+	log := rpccontext.Logger(ctx)
+
+	if len(req.Audience) == 0 {
```

Comment:
This is also checked by `mintJWTSVID`. Is this intentional?

Reply:
Double-checking was not intentional. I'll remove this check.